### ISSUE TITLE
Problem with importing "AppDelegate.h" on iOS

### ISSUE THREAD

**Issue by demoran23** [FormidableLabs/react-native-app-auth#105]
My `AppDelegate.h` file contains the following import:

> #import <RCTOneSignal.h>

This import starting giving me a `file not found` error when react-native-app-auth is built.

```
In file included from /Users/adam/Source/Repos/mobile/node_modules/react-native-app-auth/ios/RNAppAuth.m:6:
In file included from ../../../ios/StarterKit/AppDelegate.h:11:
../../../ios/StarterKit/../../node_modules/react-native-onesignal/ios/RCTOneSignal/RCTOneSignal.h:5:9: fatal error: 'OneSignal.h' file not found
#import "OneSignal.h"
        ^~~~~~~~~~~~~
1 error generated.
```

I believe this is the result of a failure to resolve the reference to `<RCTOneSignal.h>` while importing react-native's AppDelegate header file from within `RNAppAuth.m`.  As such, I updated my import statement to: 

> #import "../../node_modules/react-native-onesignal/ios/RCTOneSignal/RCTOneSignal.h"

And this eventually allowed me to build.  However, I had to edit a file within the OneSignal project itself to allow me to do so, since within the context of RNAppAuth.h it couldn't find its own imports!

I'm no master of XCode.  Is there something we can do to inherit the imports defined in the react-native project from within react-native-app-auth?

**Comment by csath** [comment]
I too have the same issue with react-native-app-auth version 2.4.1. Isn't there's a better solution than #import "../../node_modules/react-native-onesignal/ios/RCTOneSignal/RCTOneSignal.h"
Because we need to edit other projects.

**Comment by skurfuerst** [comment]
Hey everybody,

I've investigated the installation of React Native dependencies a little more in-depth; and I've come to a quite clean cocoapod-based solution (I hope). I've summed up the approach in a blog post at https://sandstorm.de/de/blog/post/react-native-managing-native-dependencies-using-xcode-and-cocoapods.html .

Maybe it helps somebody :)

All the best,
Sebastian

**Comment by hakonk** [comment]
@skurfuerst As you point out in your blog post, the podspec needs patching and the `AppDelegate` header needs to be imported. I've proposed a solution for this in the following PR: #131

**Comment by kadikraman** [comment]
This should certainly now be fixed in `v3.1.0`
